system/lib/wasmfs/backends/noderawfs_root.cpp

```diff
@@ -0,0 +1,14 @@
+// Copyright 2023 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include "emscripten/wasmfs.h"
+
+namespace wasmfs {
+
+backend_t wasmfs_create_root_dir(void) {
+  return wasmfs_create_node_backend(".");
```

Comment:
My understanding is that NODERAWFS exposed `/` as the root.. i.e. you can open `/etc/passwd` if you want.

This looks like its exposing the current working directory..

Reply:
IIUC NODERAWFS is explicitly about granting the generated program complete raw access to the entire underlying filesystem (basically making of VFS layer invisible).  I think a lot of folks depend on that so we probably want to keep that behaviour around.

Reply:
In other words, perhaps `NODERAWFS` can/should be completely independent of wasmfs?

Reply:
I don't think the code can be independent though? We have a NODERAWFS impl for the old FS and now the beginnings of one for the new FS.

I think this PR might be enough, if we also have support for looking up the directory tree. I checked that now and indeed it works in the JS FS but not in WasmFS. So we would need to set the root to be the actual system root I think. Just changing this PR to do `/` instead of `.` for the root dir doesn't work though, so that needs more investigation.

Reply:
I tested with this:
```c
#include <dirent.h>
#include <stdio.h>
#include <stdlib.h>

int main() {
  struct dirent **namelist;
  int n;

  // Scan the parent dir
  n = scandir("..", &namelist, NULL, alphasort);
  printf("n: %d\n", n);
  if (n < 0)
    return 1;

  while (n--) {
    printf("name: %s\n", namelist[n]->d_name);
    free(namelist[n]);
  }
  free(namelist);

  return 0;
}
```
A test with that might be useful to add as I don't think we have any more NODERAWFS tests atm.

Reply:
The way NODERAWFS works IIUC is to completely replace/override all the old FS code.. basically taking over completely.  I think the equivalent for the new version of `NODERAWFS` might be to simply to provide an alternative implementation of the musl syscalls and avoid include wasmfs at all. 

See https://github.com/emscripten-core/emscripten/blob/6b179e3a5fe272b444f332fcbeb589c71cfa15b8/src/library_noderawfs.js#L23-L26

Reply:
> A test with that might be useful to add as I don't think we have any more NODERAWFS tests atm.

Looking at test_core.py and test_other.py I see a fair few NODERAWFS tests.. have you tried all them with WASMFS?  I would expect/hope that at least one of them would fail with your currently "mount CWD at root" approach.

Reply:
I see. I'm not sure I can think of a good way to do that in WasmFS actually. Yeah, maybe a reimplementation at the syscall layer is worth exploring, as you said.

Reply:
> Looking at test_core.py and test_other.py I see a fair few NODERAWFS tests.. have you tried all them with WASMFS?

I haven't gotten to them all yet but all those I've tried have passed so far.